internal/cmd/flags.go

```diff
@@ -45,27 +46,102 @@ var (
 		Usage:   "original reference",
 	}
 
-	FlagPluginConfig = &cli.StringSliceFlag{
+	FlagPluginConfig = &cli.StringFlag{
 		Name:    "pluginConfig",
 		Aliases: []string{"pc"},
 		Usage:   "list of comma-separated {key}={value} pairs that are passed as is to the plugin, refer plugin documentation to set appropriate values",
 	}
 )
 
-func ParseFlagPluginConfig(pluginConfigSlice []string) (map[string]string, error) {
-	if len(pluginConfigSlice) == 0 {
+// KeyValueSlice is a flag with type int
+type KeyValueSlice interface {
+	Set(value string) error
+	String() string
+}
+
+func ParseFlagPluginConfig(ctx *cli.Context) (map[string]string, error) {
+	val := ctx.String(FlagPluginConfig.Name)
+	pluginConfig, err := ParseKeyValueListFlag(val)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse %q as value for flag %s: %s", val, FlagPluginConfig.Name, err)
+	}
+	return pluginConfig, nil
+}
+
+func ParseKeyValueListFlag(val string) (map[string]string, error) {
+	if val == "" {
 		return nil, nil
 	}
-	m := make(map[string]string, len(pluginConfigSlice))
-	for _, c := range pluginConfigSlice {
-		if k, v, ok := strings.Cut(c, "="); ok {
+	flags, err := splitQuoted(val)
+	if err != nil {
+		return nil, err
+	}
+	m := make(map[string]string, len(flags))
+	for _, c := range flags {
+		if c == "" {
+			return nil, fmt.Errorf("empty entry: %q", c)
```

Comment:
This doesn't error out `a=b,=`
Let's check the key and value are not empty after space trimming them.

Reply:
Done!